### ISSUE TITLE
Error while pushing from local repo to postgis (type "objectid" does not exist)

### ISSUE THREAD

**Issue by mclucky** [locationtech/geogig#438]
Hi!

Workflow is the following:

1. created an empty repo in postgis
2. cloned into a local repo
3. imported one shape file with one feature
4. add, commit, push

Then there is an error while pushing - see the attached log file.

[geogig.log](https://github.com/locationtech/geogig/files/2103942/geogig.log)

Does anyone have an idea?

**Comment by mclucky** [comment]
PFA the shp file I've used

[poly.zip](https://github.com/locationtech/geogig/files/2103948/poly.zip)

**Comment by emerkle826** [comment]
@mclucky 
I just pulled down you Shapefile and tried the following steps. NOTE: the database must exist in PostgreSQL before you can use GeoGig to initialize a PostgreSQL repo.

To create my database, I logged into my local PostgreSQL and did this:
```sh
create DATABASE repos OWNER geogig;
```
Now I can use `postgresql://localhost:5432/repos/public/repo1?user=geogig&password=geogig` as my repository connection URI.

Initialize a GeoGig repository in PostgreSQL (you don't need the PostGIS extension, plain PostgreSQL is fine):
```sh
geogig --repo 'postgresql://localhost:5432/repos/public/repo1?user=geogig&password=geogig' init
```
Clone the PostgreSQL repo to a local repository (directory):
```sh
geogig clone 'postgresql://localhost:5432/repos/public/repo1?user=geogig&password=geogig' ./clone
```
Change into the local repo directory
```sh
cd clone
```
Configure local repository username and email:
```sh
geogig config user.name erik
geogig config user.email erik
```
Import the shapefile (I unzipped it in the parent directory of my clone)
```sh
geogig shp import ../poly.shp
```
Add the import to the staging area and commit
```sh
geogig add
geogig commit -m "initial commit"
```
Push local changes to the remote PostgreSQL repo:
```sh
geogig push origin
```
This worked just fine (I haven't actually done this in a long time personally).

If these steps do not work for you, please provide:
- GeoGig version (geogig --version output)
- PostgreSQL version
- Java version (java --version output)
- O/S

**Comment by mclucky** [comment]
Hi @emerkle826 

Thx for your reply. This is exactly what I've done, except that I haven't created the postgis repo myself. I have to talk to my fellow to clarify if he had done everything the right way.

Sorry that I did not provide the following details in the first place.

Java
1.8.0_171 (runtime environment)

GeoGig CLI
1.2.0
beff35ead822efb373f9125eef9e608e8dcfa342

PostgreSQL
9.4.8

OS
Windows 10

But let me check next week, I'll give it another try.

**Comment by DBlasby** [comment]
Hi,

I saw this in the log;

Caused by: org.postgresql.util.PSQLException: ERROR: type "objectid" does
not exist


This indicates that the geogig database doesn't have the "objectid" type
(used to store the SHA hash).  I don't know how this is possible - how did
you init your geogig database?

Dave

**Comment by pinkerltm** [comment]
Hi! I have initialized the database for @mclucky , mas be likely that I did something wrong, as I first initialized it via the SQL Script which is published in the geogig documentation. After reading that geogig creates these tables by itself if they are not existing, I dropped everything and started over by the geogig init command you mentioned above.

I was wondering that all tables were created in the public schema while only the objectid type was created in the users default schema

**Comment by pinkerltm** [comment]
@emerkle826 Thanks for your walkthrough, I dropped all geogig objects I initialized, then I reinitialized it the way you mentioned.

The Problem was 
`geogig --repo 'postgresql://localhost:5432/repos/public/repo1?user=geogig&password=geogig' init`

which had to be simply modified to

`geogig --repo "postgresql://localhost:5432/repos/public/repo1?user=geogig&password=geogig" init` 

on Windows 10 (CMD, didn't try on PS yet). Then the repository got initialized as expected and all other steps also work.

Thanks for giving advice!
